File: sdk-python/src/beamlit/authentication/apikey.py
from typing import Generator

from httpx import Auth, Request, Response


class ApiKeyProvider(Auth):
    def __init__(self, credentials, workspace_name: str):
        self.credentials = credentials
        self.workspace_name = workspace_name

    def auth_flow(self, request: Request) -> Generator[Request, Response, None]:
        request.headers['X-Beamlit-Api-Key'] = self.credentials.api_key
        request.headers['X-Beamlit-Workspace'] = self.workspace_name
        yield request

File: sdk-ts/src/authentication/apikey.ts
import { Credentials } from "./types.js";

/**
 * Handles API key authentication for Blaxel API requests.
 */
export class ApiKeyAuth {
  private credentials: Credentials;
  private workspaceName: string;

  /**
   * Constructs a new ApiKeyAuth instance.
   * @param credentials - The credentials containing the API key.
   * @param workspaceName - The name of the workspace.
   */
  constructor(credentials: Credentials, workspaceName: string) {
    this.credentials = credentials;
    this.workspaceName = workspaceName;
  }

  /**
   * Retrieves the authentication headers.
   * @returns A promise resolving to a record of header key-value pairs.
   */
  async getHeaders(): Promise<Record<string, string>> {
    return {
      "X-Blaxel-Api-Key": this.credentials.apiKey || "",
      // "X-Blaxel-Authorization": `Bearer ${this.credentials.apiKey || ""}`,
      "X-Blaxel-Workspace": this.workspaceName,
    };
  }

  /**
   * Intercepts and sets the authentication headers on a request.
   * @param req - The request to intercept and modify.
   */
  intercept(req: Request): void {
    req.headers.set("X-Blaxel-Api-Key", this.credentials.apiKey || "");
    req.headers.set("X-Blaxel-Workspace", this.workspaceName);
  }
}
